### DISCUSSION TITLE
How to run in VSCode in debug mode?

### DISCUSSION THREAD

**Discussion by scheung38** [cookiecutter/cookiecutter-django#3696] [Q&A]
This demo shows how to setup 

docker run -v ${PWD}/app:/app -w /app python:3.9-alpine sh -c "pip install Django==3.2 && django-admin startproject app ."

Debugging a Dockerized Django app with VSCode
https://www.youtube.com/watch?v=x7lZAmMVo2M

But it is a standalone Django app, so like to ask what do we need to configure to run in debug mode

Also docker-compose -f local.yml run --rm --service-ports django
Error response from daemon: driver failed programming external connectivity on endpoint arthur_paf_django_run_e7f6875bf0dd (1b2d65790fa922b927d77fe3020800ed88195ebec295610aaa273cf7e160d1a9): Bind for 0.0.0.0:8000 failed: port is already allocated

But prior to running this command I do not have other processes

**Comment by scheung38** [comment]
Followed this tutorial but not sure why VSCode debugger does not attach https://testdriven.io/blog/django-debugging-vs-code/ 

Then https://code.visualstudio.com/docs/containers/debug-python 

but cookie-cutter does not have task.json?

**Reply by browniebroke** [reply]
> cookie-cutter does not have task.json?

No, this template doesn't have this. I'm not using VSCode so I have no personal interest in this, however I'm not against adding it, VSCode is popular enough to justify such addition. 

If you manage to make it work on your project, feel free to share your config here, or even better, send a PR to add it to the template. At this point, you'll probably have done the hardest bit 😃 

This is a community based project, things are not added until someone takes the time to implement it.
